api_tests/users/serializers/test_serializers.py

```diff
@@ -1,45 +1,181 @@
 import pytest
-import mock
 
 from api.users.serializers import UserSerializer
+from api_tests.utils import create_test_file
+
 from osf_tests.factories import (
     UserFactory,
+    RegistrationFactory,
+    WithdrawnRegistrationFactory,
+    PreprintFactory,
+    ProjectFactory,
+    InstitutionFactory
 )
 from tests.utils import make_drf_request_with_version
+from django.utils import timezone
+from django.urls import resolve, reverse
 
+from osf.models import QuickFilesNode
 
 @pytest.fixture()
-@mock.patch('website.search.elastic_search.update_user')
-def user(mock_update_user):
+def user():
     user = UserFactory()
-    user.jobs = [{
-        'title': 'Veterinarian/Owner',
-        'ongoing': True,
-        'startYear': '2009',
-        'startMonth': 4,
-        'institution': 'Happy Paws Vet'
-    }]
-    user.schools = [{
-        'endYear': '1994',
-        'ongoing': False,
-        'endMonth': 6,
-        'startYear': '1990',
-        'department': 'Veterinary Medicine',
-        'startMonth': 8,
-        'institution': 'UC Davis'
-    }]
-    user.save()
+    quickfiles_node = QuickFilesNode.objects.get_for_user(user)
+    create_test_file(quickfiles_node, user)
+    inst = InstitutionFactory()
+    user.affiliated_institutions.add(inst)
     return user
 
 
+@pytest.fixture()
+def user_without_nodes():
+    return UserFactory()
+
+
+@pytest.fixture()
+def preprint(user):
+    return PreprintFactory(creator=user)
+
+
+@pytest.fixture()
+def unpublished_preprint(user):
+    return PreprintFactory(is_published=False, creator=user)
+
+
+@pytest.fixture()
+def deleted_preprint(user):
+    preprint = PreprintFactory(creator=user)
+    preprint.deleted = timezone.now()
+    preprint.save()
+    return preprint
+
+
+@pytest.fixture()
+def withdrawn_preprint(user):
+    preprint = PreprintFactory(is_published=False, creator=user)
+    preprint.date_withdrawn = timezone.now()
+    preprint.save()
+    return preprint
+
+
+@pytest.fixture()
+def private_preprint(user):
+    preprint = PreprintFactory(creator=user)
+    preprint.is_public = False
+    preprint.save()
+    return preprint
+
+
+@pytest.fixture()
+def registration(user):
+    return RegistrationFactory(creator=user, is_public=True)
+
+
+@pytest.fixture()
+def withdrawn_registration(registration):
+    return WithdrawnRegistrationFactory(registration=registration, user=registration.creator)
+
+
+@pytest.fixture()
+def private_registration(user):
+    return RegistrationFactory(creator=user, is_public=False)
+
+
+@pytest.fixture()
+def project(user):
+    return ProjectFactory(creator=user)
+
+
+@pytest.fixture()
+def public_project(user):
+    return ProjectFactory(creator=user, is_public=True)
+
+
+@pytest.fixture()
+def deleted_project(user):
+    return ProjectFactory(creator=user, is_deleted=True)
+
+
+def pytest_generate_tests(metafunc):
```

Comment:
I'm sorry this might seem over-engineered but it provides a really readable concise test suite that's hard to pass up.

Reply:
it's great for scenarios like this 👍